ktor-core/src/org/jetbrains/ktor/application/ApplicationResponse.kt

```diff
@@ -20,6 +19,31 @@ public interface ApplicationResponse {
     public fun interceptSend(handler: (message: Any, next: (message: Any) -> ApplicationRequestStatus) -> ApplicationRequestStatus)
 }
 
+public abstract class ResponseHeaders {
+    private val headersChain = Interceptable2<String, String, Unit> { name, value ->
+        appendHeaderImpl(name, value)
+    }
+
+    public fun contains(name: String): Boolean = getHeaderValuesImpl(name).isNotEmpty()
+    public fun get(name: String): String? = getHeaderValuesImpl(name).firstOrNull()
```

Comment:
singleOrNull, probably? What is intended behaviour when we append multiple headers, and the get one?